src/util/secrets/secrets.c

```diff
@@ -1006,10 +1056,15 @@ errno_t sss_sec_get(TALLOC_CTX *mem_ctx,
     attr_enctype = ldb_msg_find_attr_as_string(res->msgs[0], "enctype", NULL);
 
     if (attr_enctype) {
-        ret = local_decrypt(req->sctx, mem_ctx, attr_secret, attr_enctype, _secret);
+        enctype = sss_sec_str_to_enctype(attr_enctype);
+        ret = local_decrypt(req->sctx, mem_ctx, attr_secret, enctype, _secret);
         if (ret) goto done;
     } else {
```

Comment:
Isn't this an error? Is it expected that this attr can be "legally" missing?

Reply:
Other approach could be to let `sss_sec_str_to_enctype(NULL) = SSS_SEC_PLAINTEXT` and to not bother with if-else at all.

Reply:
Right, good suggestion. AFAIK it should not be missing but defaulting to "plaintext" is a good fallback.